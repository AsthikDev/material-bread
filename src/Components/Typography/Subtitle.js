import React, { Component } from 'react';
import PropTypes from 'prop-types';
import BaseText from './BaseText';
import withTheme from '../../Theme/withTheme';

class Subtitle extends Component {
  static propTypes = {
    type: PropTypes.number,
    children: PropTypes.node,
    theme: PropTypes.object,
  };

  numberToStyle(num) {
    const { theme } = this.props;
    const numArray = [theme.subtitleOne, theme.subtitleTwo];
    return numArray[num - 1];
  }

  render() {
    const { type, theme } = this.props;
    let textStyle;
    if (type < 1 && type > 2) {
      textStyle = theme.subtitleOne;
    }
    textStyle = this.numberToStyle(type);

    return (
      <BaseText typographyStyles={textStyle} {...this.props}>
        {this.props.children}
      </BaseText>
    );
  }
}

export default withTheme(Subtitle);
