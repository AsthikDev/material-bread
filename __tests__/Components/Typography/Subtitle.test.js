import React from 'react';
import Subtitle from '../../../src/Components/Typography/Subtitle';

import renderer from 'react-test-renderer';

test('it renders Subtitle One', () => {
  const tree = renderer.create(<Subtitle type={1}>Subtitle</Subtitle>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('it renders Subtitle Two', () => {
  const tree = renderer.create(<Subtitle type={2}>Subtitle</Subtitle>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('it renders Subtitle One when a value isnt 1 or 2', () => {
  const tree = renderer.create(<Subtitle type={3}>Subtitle</Subtitle>).toJSON();
  expect(tree).toMatchSnapshot();
});
