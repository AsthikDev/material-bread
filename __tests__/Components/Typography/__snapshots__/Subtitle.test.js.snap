// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders Subtitle One 1`] = `
<Text
  style={
    Object {
      "color": "rgba(0,0,0, 87)",
      "fontSize": 16,
      "letterSpacing": 0.15,
      "lineHeight": 16,
      "marginBottom": 0,
      "textAlign": "left",
    }
  }
  theme={
    Object {
      "bodyText": Object {
        "fontSize": 16,
        "letterSpacing": 0.5,
        "lineHeight": 16,
      },
      "bodyTextTwo": Object {
        "fontSize": 14,
        "letterSpacing": 0.25,
        "lineHeight": 14,
      },
      "button": Object {
        "borderRadius": 4,
        "color": "#2196f3",
        "height": 36,
        "minHeight": 36,
        "padding": 8,
      },
      "buttonText": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 1.25,
      },
      "caption": Object {
        "fontSize": 12,
        "letterSpacing": 0.4,
        "lineHeight": 12,
      },
      "divider": Object {
        "backgroundColor": "black",
        "height": 0.5,
        "marginVertical": 16,
        "opacity": 0.12,
        "width": "100%",
      },
      "headingFive": Object {
        "fontSize": 24,
        "lineHeight": 24,
      },
      "headingFour": Object {
        "fontSize": 34,
        "letterSpacing": 0.25,
        "lineHeight": 34,
      },
      "headingOne": Object {
        "fontSize": 96,
        "fontWeight": "300",
        "letterSpacing": -1.5,
        "lineHeight": 96,
      },
      "headingSix": Object {
        "fontSize": 20,
        "fontWeight": "500",
        "letterSpacing": 0.15,
        "lineHeight": 20,
      },
      "headingThree": Object {
        "fontSize": 48,
        "lineHeight": 48,
      },
      "headingTwo": Object {
        "fontSize": 60,
        "fontWeight": "300",
        "letterSpacing": -0.5,
        "lineHeight": 60,
      },
      "outlinedButton": Object {
        "borderColor": "rgba(0,0,0,.29)",
        "borderWidth": 0.5,
      },
      "overline": Object {
        "fontSize": 10,
        "letterSpacing": 1.5,
        "lineHeight": 10,
        "textTransform": "uppercase",
      },
      "paper": Object {
        "backgroundColor": "white",
        "borderRadius": 2,
      },
      "subtitleOne": Object {
        "fontSize": 16,
        "letterSpacing": 0.15,
        "lineHeight": 16,
      },
      "subtitleTwo": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 0.1,
        "lineHeight": 14,
      },
    }
  }
  type={1}
  typographyStyles={
    Object {
      "fontSize": 16,
      "letterSpacing": 0.15,
      "lineHeight": 16,
    }
  }
>
  Subtitle
</Text>
`;

exports[`it renders Subtitle One when a value isnt 1 or 2 1`] = `
<Text
  style={
    Object {
      "color": "rgba(0,0,0, 87)",
      "marginBottom": 0,
      "textAlign": "left",
    }
  }
  theme={
    Object {
      "bodyText": Object {
        "fontSize": 16,
        "letterSpacing": 0.5,
        "lineHeight": 16,
      },
      "bodyTextTwo": Object {
        "fontSize": 14,
        "letterSpacing": 0.25,
        "lineHeight": 14,
      },
      "button": Object {
        "borderRadius": 4,
        "color": "#2196f3",
        "height": 36,
        "minHeight": 36,
        "padding": 8,
      },
      "buttonText": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 1.25,
      },
      "caption": Object {
        "fontSize": 12,
        "letterSpacing": 0.4,
        "lineHeight": 12,
      },
      "divider": Object {
        "backgroundColor": "black",
        "height": 0.5,
        "marginVertical": 16,
        "opacity": 0.12,
        "width": "100%",
      },
      "headingFive": Object {
        "fontSize": 24,
        "lineHeight": 24,
      },
      "headingFour": Object {
        "fontSize": 34,
        "letterSpacing": 0.25,
        "lineHeight": 34,
      },
      "headingOne": Object {
        "fontSize": 96,
        "fontWeight": "300",
        "letterSpacing": -1.5,
        "lineHeight": 96,
      },
      "headingSix": Object {
        "fontSize": 20,
        "fontWeight": "500",
        "letterSpacing": 0.15,
        "lineHeight": 20,
      },
      "headingThree": Object {
        "fontSize": 48,
        "lineHeight": 48,
      },
      "headingTwo": Object {
        "fontSize": 60,
        "fontWeight": "300",
        "letterSpacing": -0.5,
        "lineHeight": 60,
      },
      "outlinedButton": Object {
        "borderColor": "rgba(0,0,0,.29)",
        "borderWidth": 0.5,
      },
      "overline": Object {
        "fontSize": 10,
        "letterSpacing": 1.5,
        "lineHeight": 10,
        "textTransform": "uppercase",
      },
      "paper": Object {
        "backgroundColor": "white",
        "borderRadius": 2,
      },
      "subtitleOne": Object {
        "fontSize": 16,
        "letterSpacing": 0.15,
        "lineHeight": 16,
      },
      "subtitleTwo": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 0.1,
        "lineHeight": 14,
      },
    }
  }
  type={3}
>
  Subtitle
</Text>
`;

exports[`it renders Subtitle Two 1`] = `
<Text
  style={
    Object {
      "color": "rgba(0,0,0, 87)",
      "fontSize": 14,
      "fontWeight": "500",
      "letterSpacing": 0.1,
      "lineHeight": 14,
      "marginBottom": 0,
      "textAlign": "left",
    }
  }
  theme={
    Object {
      "bodyText": Object {
        "fontSize": 16,
        "letterSpacing": 0.5,
        "lineHeight": 16,
      },
      "bodyTextTwo": Object {
        "fontSize": 14,
        "letterSpacing": 0.25,
        "lineHeight": 14,
      },
      "button": Object {
        "borderRadius": 4,
        "color": "#2196f3",
        "height": 36,
        "minHeight": 36,
        "padding": 8,
      },
      "buttonText": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 1.25,
      },
      "caption": Object {
        "fontSize": 12,
        "letterSpacing": 0.4,
        "lineHeight": 12,
      },
      "divider": Object {
        "backgroundColor": "black",
        "height": 0.5,
        "marginVertical": 16,
        "opacity": 0.12,
        "width": "100%",
      },
      "headingFive": Object {
        "fontSize": 24,
        "lineHeight": 24,
      },
      "headingFour": Object {
        "fontSize": 34,
        "letterSpacing": 0.25,
        "lineHeight": 34,
      },
      "headingOne": Object {
        "fontSize": 96,
        "fontWeight": "300",
        "letterSpacing": -1.5,
        "lineHeight": 96,
      },
      "headingSix": Object {
        "fontSize": 20,
        "fontWeight": "500",
        "letterSpacing": 0.15,
        "lineHeight": 20,
      },
      "headingThree": Object {
        "fontSize": 48,
        "lineHeight": 48,
      },
      "headingTwo": Object {
        "fontSize": 60,
        "fontWeight": "300",
        "letterSpacing": -0.5,
        "lineHeight": 60,
      },
      "outlinedButton": Object {
        "borderColor": "rgba(0,0,0,.29)",
        "borderWidth": 0.5,
      },
      "overline": Object {
        "fontSize": 10,
        "letterSpacing": 1.5,
        "lineHeight": 10,
        "textTransform": "uppercase",
      },
      "paper": Object {
        "backgroundColor": "white",
        "borderRadius": 2,
      },
      "subtitleOne": Object {
        "fontSize": 16,
        "letterSpacing": 0.15,
        "lineHeight": 16,
      },
      "subtitleTwo": Object {
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 0.1,
        "lineHeight": 14,
      },
    }
  }
  type={2}
  typographyStyles={
    Object {
      "fontSize": 14,
      "fontWeight": "500",
      "letterSpacing": 0.1,
      "lineHeight": 14,
    }
  }
>
  Subtitle
</Text>
`;
